src/core/agent/services/identifierService.ts

```diff
@@ -303,6 +313,150 @@ class IdentifierService extends AgentService {
+    if (!notifications.length) {
+      return false;
+    }
+    const exn = notifications[0].exn;
+    const signifyName = exn.a.name;
+    try {
+      const multiSig = await this.agent.modules.signify.getIdentifierByName(
+        signifyName
+      );
+      if (multiSig) {
+        return true;
+      }
+    } catch (e) {
+      return false;
+    }
+    return false;
+  }
+
+  async joinMultisig(
+    notificationId: string,
+    meta: Pick<
+      IdentifierMetadataRecordProps,
+      "displayName" | "colors" | "theme"
+    >
+  ): Promise<string | undefined> {
+    const savedMultisigNotification = await this.agent.genericRecords.findById(
+      notificationId
+    );
+    if (!savedMultisigNotification) {
+      return;
+    }
+    const msgSaid = savedMultisigNotification?.content?.d as string;
+    const hasJoined = await this.hasJoinedMultisig(msgSaid); //"EKHJNmH2lyRL1RRNjeG3hzbcuimypSVpEFC8hVDtcr6F"
+    if (hasJoined) {
+      await this.agent.genericRecords.deleteById(
+        savedMultisigNotification?.id as string
+      );
+      return;
+    }
+    const notifications: MultiSigIcpNotification[] =
+      await this.agent.modules.signify.getNotificationsBySaid(msgSaid);
+
+    if (!notifications.length) {
+      return;
+    }
+    const exn = notifications[0].exn;
+    const rstate = exn.a.rstates;
+    const identifiers = await this.getIdentifiers();
+    const identifier = identifiers.find((identifier) => {
+      return rstate.find((item) => identifier.id === item.i);
+    });
+
+    if (identifier && identifier.signifyName) {
+      const aid = await this.agent.modules.signify.getIdentifierByName(
+        identifier?.signifyName
+      );
+      const res = await this.agent.modules.signify.joinMultisig(exn, aid);
+      await this.agent.genericRecords.deleteById(
+        savedMultisigNotification?.id as string
+      );
+      const multisigId = res.op.name.split(".")[1];
+      await this.createIdentifierMetadataRecord({
+        id: multisigId,
+        displayName: meta.displayName,
+        method: IdentifierType.KERI,
+        colors: meta.colors,
+        theme: meta.theme,
+        signifyName: utils.uuid(),
+        opName: res.op.name, //we save the opName here to sync the multisig's status later
+        isPending: res.op.done ? false : true, //this will be updated once the operation is done
+      });
+      return multisigId;
+    }
+  }
+
+  async updateMultisigMetadata(metadata: IdentifierMetadataRecord) {
+    if (!metadata.opName || !metadata.isPending) {
+      return;
+    }
+    const pendingOperation = await this.agent.modules.signify.getOpByName(
+      metadata.opName
+    );
+    if (pendingOperation && pendingOperation.done) {
+      await this.agent.modules.generalStorage.updateIdentifierMetadata(
+        metadata.id,
+        { isPending: false }
+      );
+    }
+  }
+
+  async getUnhandledMultisigIdentifiers(): Promise<KeriNotification[]> {
```

Comment:
When will this be used?

Reply:
This will be called when the user starts the app. We will need to handle these later in the UI.